src/Host/Client/Impl/Definitions/IMessageTransport.cs

```diff
@@ -9,5 +9,6 @@ namespace Microsoft.R.Host.Client {
     public interface IMessageTransport {
         Task SendAsync(string message, CancellationToken ct = default(CancellationToken));
         Task<string> ReceiveAsync(CancellationToken ct = default(CancellationToken));
+        Task<byte[]> ReceiveRawAsync(CancellationToken ct = default(CancellationToken));
```

Comment:
I think we should revisit this. This interface is too low-level now - it's meant to be the message transport, but it's effectively a packet transport now (since you need multiple `Receive` calls to get a single message where blobs are involved).

Let's move `Message` and all related functionality here, so that `RHost` only ever deals with complete message instances. This will also let us do things more efficiently if we ever decide to move to a different protocol that allows mixing binary and non-binary data in a more efficient way.